torch/nn/modules/loss.py

```diff
@@ -134,9 +135,10 @@ class NLLLoss(_WeightedLoss):
            class. If given, it has to be a Tensor of size `C`. Otherwise, it is
            treated as if having all ones.
         size_average (bool, optional): By default, the losses are averaged
-           over observations for each minibatch with weights set by
-           :attr:`weight`. However, if the field :attr:`size_average` is set to
+           over each loss element in the batch. Note that for some losses, there
```

Comment:
It might be better to have two versions of the description to clarify if the loss does an average over all elements, or if the loss averages over batch_size.  Users might not understand "losses are averaged over each loss element" at first glance